src/sentry/tasks/base.py

```diff
@@ -44,10 +49,17 @@ def _wrapped(*args, **kwargs):
             with metrics.timer(key, instance=instance), \
                     track_memory_usage('jobs.memory_change', instance=instance):
                 try:
-                    result = func(*args, **kwargs)
+                    return func(*args, **kwargs)
+                except SoftTimeLimitExceeded as error:
+                    logger.warning(
```

Comment:
I was wondering the same thing myself. I don't think we use retries anywhere that we also use timeouts, so this should work for now. Ideally, this would just patch the context with the fingerprint before reraising — I think that just setting `self.context.merge({'fingerprint': [type(error).__name__, instance]})` would work?